test/integration/runner-spec/iterationCount.test.js

```diff
@@ -0,0 +1,112 @@
+describe('Run with iterationCount set', function () {
+    describe('without data', function () {
+        var testrun;
+
+        before(function (done) {
+            this.run({
+                iterationCount: 3,
+                collection: {
+                    item: [{
+                        event: [{
+                            listen: 'prerequest',
+                            script: {
+                                exec: `
+                                    pm.test('should contain data', function () {
+                                        pm.expect(pm.variables.get('foo')).to.equal('bar');
+                                    });
+                                `
+                            }
+                        }],
+                        request: {
+                            url: 'https://postman-echo.com/get',
+                            method: 'GET'
+                        }
+                    }]
+                }
+            }, function (err, results) {
+                testrun = results;
+                done(err);
+            });
+        });
+
+        it('should run collection specified number of times', function () {
+            expect(testrun).to.be.ok;
+            expect(testrun.done.getCall(0).args[0]).to.not.exist;
+            expect(testrun).to.nested.include({
+                'done.calledOnce': true,
+                'start.calledOnce': true
+            });
+
+            expect(testrun.request.callCount).to.equal(3);
+        });
+    });
+
+    describe('with data items lesser than iteration count', function () {
```

Comment:
@vikiCoder ensure we test for:
- data.length > iterationCount
- data.length < iterationCount
- data without iterationCount